### PR TITLE
Add handling for config files

### LINKED ISSUE #12: Create Files if imgur and reddit info not found
If imgurInfo and redditInfo are not found, take user input and ask if they'd like to save it. Use getpass for this.

### LINKED ISSUE #30: File modes should be more strict
Files should be opened using a mode that indicates what's going to be done with them.
A large offender that I found immediately as I started looking as lines `35-38`, which contain `"a+"` when the only thing that's being done with the file is reading from it.

In general, use the most restrictive possible format that you can. (`"r"` for something you only read from, `"a"` if you're appending but not reading, etc.)

### LINKED ISSUE #38: Improve code readability.
This would include better pep8 compliance, reduction of large ternary clauses, and probably moving more things into functions.

I can take this on.

### LINKED ISSUE #51: Define bot Account User Name as Variable
Just something like:
botUser = "foobar_example123"

It should be a string.
Replace all instances of "DEAL_WITH_IT_bot" with botUser or whatever the var is named.  

This will help the flexibility issues @porglezomp mentioned in #48.

### PR DESCRIPTION
**Pull request by porglezomp**
Created module `config`
Change bot to use config.txt
Moved some functions into their own modules (`karma` and `dankutil`)
Improved PEP8 complaince (but with max 99 character lines)

In this PR:
Fix #30
Fix #51
Fix #38 
Fix #12

### CODE REVIEW

File: bot.py [hunk 1]
```diff
@@ -31,49 +34,26 @@
...
 auth.secure = True
-auth.set_access_token(access_token, access_token_secret)
-
-# Create blacklist_subs.txt if it doesn't exist
-if not os.path.isfile("blacklist_subs.txt"):
-    open("blacklist_subs.txt", "a").close()
+auth.set_access_token(conf['twitter_access_token'], conf['twitter_access_token_secret'])
 
 # Get all the blacklisted subs, put them into a list
+dankutil.ensure_file_exists("balcklist_subs.txt")
```
Comment (powderblock) [inline bot.py]: balcklist_subs.txt should be blacklist_sub.txt

**Comment by powderblock** [comment]
balcklist_subs.txt on line 50 is not spelled correctly. (It should be blacklist_subs.txt)

**Comment by powderblock** [comment]
After this line you run over the loaded post data in a for loop:

# File to load post IDs from
postsFile = open("posts.txt", "a+")

(I would comment on the line but somehow it's not showing up in the diff.)

Then you load the posts but you use the wrong(?) var name? It's weird.
The line is something like:
for post in range(0, len(postItems)):
    already_done.add(str(postItems[post]))

But postItems is not defined.
 I need help on this, talk to me privately. I am having too many issues to merge this.

**Comment by porglezomp** [comment]
Okay, I'll look into all those issues and make another commit.

**Comment by porglezomp** [comment]
Yeah, I just screwed up there. You did some other work right in the middle of the initialization, so it was kind of hard to follow, and I just read it wrong. Anyway, I'll commit those fixes in a moment.

**Comment by porglezomp** [comment]
See if that fixes it.

**Comment by powderblock** [comment]
# After changing line 189 (see below) the client now seems to be working.

### Two main questions / issues:
-On line 76 in bot.py, you print the message template, this is useful for debugging but in practice is not needed. Did you leave this in by mistake?


-On line 189 in bot.py, you log into reddit using the vars username and password that have been extracted from conf, this is great, except that the values we extracted out of config.txt are called reddit_username and reddit_password. (It's clear you understand this, reddit_username is used on the line before (line 188 in bot.py.)). I had to change this line to use reddit_username and reddit_password, but after that change, it worked!  

Good job! I am going to let this run overnight to check for bugs and will merge in the morning if all is well. (Fix issue 2 with a new commit, the first issue is not huge but it looks annoying on screen.)

Thanks again!

**Comment by powderblock** [comment]
Keeps crashing with this error:
Traceback (most recent call last):
  File "bot.py", line 203, in <module>
    postsFile.write(post.id + "\n")
IOError: [Errno 0] Error

What file mode do we open postsFile with?

**Comment by porglezomp** [comment]
One interesting thing, I couldn't possibly have found that error on my own, because it's a limitation of how file handling has to work on Windows. (See http://stackoverflow.com/questions/11176724/python-file-operations)

Anyway, I'm not just opening and closing the file every time you want to deal with it.

**Comment by powderblock** [comment]
Seems like the major things are fixed. Testing.

Thank you.

**Comment by powderblock** [comment]
On line 116, client_id needs to be imgur_client_id.
im = pyimgur.Imgur(client_id) into im = pyimgur.Imgur(imgur_client_id)

**Comment by powderblock** [comment]
### This is wrong. imgur_client_id does not work. 

The error I get when trying to use imgur_client_id is "global name 'imgur_client_id' is not defined". You define it as imgur_client_id at the beginning of the file, but then refer it to as just client_id later. Are you maybe just defining it locally in config.py? (I think this is where you assign the values?) Or are you assigning it at all?

#### This is weird because imgur_client_id should work.

**Comment by porglezomp** [comment]
It needs to be conf['imgur_client_id']. I'll fix.

**Comment by porglezomp** [comment]
Actually, if you can make that fix on your own and test more before I push again, that'd be nice.

**Comment by powderblock** [comment]
I made that change.

Testing.

**Comment by powderblock** [comment]
Working so far.

**Comment by porglezomp** [comment]
Okay, I'll push that, it sounds like you can merge it.

**Comment by powderblock** [comment]
I don't know why. But it ran all night without making a single post.

I am going to merge and assume it was a fluke, but I am bit uneasy.
I am going to watch it pretty carefully for the next 24 hours.

**Comment by porglezomp** [comment]
No debug output indicating anything?

**Comment by powderblock** [comment]
No. It was just like it never ran the next loop to check for new images.

**Comment by powderblock** [comment]
It ran fine for like 45 minutes and made one post, but after that it never made another one.

**Comment by powderblock** [comment]
After restarting it worked fine. (It is current what is powering @DealWithItbot's Twitter)

**Comment by porglezomp** [comment]
Add more prints for better debugging so you can see where it's going wrong.
